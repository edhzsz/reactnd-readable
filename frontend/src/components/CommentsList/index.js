import React from 'react';
import PropTypes from 'prop-types';
import './CommentsList.css'
import Comment from '../Comment'

const CommentsList = (props) => {
  return (
    <div className="comments-list">
      <span>Comments</span>
      <div>
        {props.comments.map(comment => (<Comment comment={comment} key={comment.id} />))}
      </div>
    </div>
  );
}

CommentsList.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    timestamp: PropTypes.number,
    voteScore: PropTypes.number,
  }))
}

export default CommentsList;