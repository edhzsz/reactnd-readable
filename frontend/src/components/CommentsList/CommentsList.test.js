import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommentsList from '.';

const testComments = [
  {
    id: '894tuq4ut84ut8v4t8wun89g',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1468166872634,
    body: 'Hi there! I am a COMMENT.',
    author: 'thingtwo',
    voteScore: 6,
    deleted: false,
    parentDeleted: false
  },
  {
    id: '8tu4bsun805n8un48ve89',
    parentId: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1469479767190,
    body: 'Comments. Are. Cool.',
    author: 'thingone',
    voteScore: -5,
    deleted: false,
    parentDeleted: false
  }
];

const wrapper = shallow(<CommentsList comments={testComments}/>);

describe('Component: CommentsList', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CommentsList />, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});