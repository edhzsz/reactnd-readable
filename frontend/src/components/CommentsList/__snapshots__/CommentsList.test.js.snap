// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CommentsList should match its snapshot with required props 1`] = `
<div
  className="comments-list"
>
  <span>
    Comments
  </span>
  <div>
    <Comment
      comment={
        Object {
          "author": "thingtwo",
          "body": "Hi there! I am a COMMENT.",
          "deleted": false,
          "id": "894tuq4ut84ut8v4t8wun89g",
          "parentDeleted": false,
          "parentId": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1468166872634,
          "voteScore": 6,
        }
      }
      key="894tuq4ut84ut8v4t8wun89g"
    />
    <Comment
      comment={
        Object {
          "author": "thingone",
          "body": "Comments. Are. Cool.",
          "deleted": false,
          "id": "8tu4bsun805n8un48ve89",
          "parentDeleted": false,
          "parentId": "8xf0y6ziyjabvozdd253nd",
          "timestamp": 1469479767190,
          "voteScore": -5,
        }
      }
      key="8tu4bsun805n8un48ve89"
    />
  </div>
</div>
`;
